File: ecmascript/rna-transcription/rna-transcription.js
const DNA_TO_RNA = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
};

class Transcriptor {
  toRna(dna_strand) {
    return dna_strand.split('')
                     .map(x => DNA_TO_RNA[x])
                     .join('')
  }
}

export default Transcriptor;

File: nim/rna-transcription/rna_transcription.nim
import sequtils
import strutils
import tables

# I wish you could just cast a char to a string in Nim.
proc toString(ch: char): string =
  repeat(ch, 1)

const dnaToRna = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}.toTable

proc translate(c: char): string = dnaToRna[c].toString

proc toRna*(dna: string): string = toSeq(dna.items).map(translate).join
